reactor-netty5-core/src/test/java/reactor/netty5/NettyOutboundTest.java

```diff
@@ -163,6 +163,7 @@ protected void encodeAndClose(ChannelHandlerContext ctx, Buffer msg,
 							List<Object> out) {
 						clearMessages.add(msg.readCharSequence(msg.readableBytes(), StandardCharsets.UTF_8));
 						out.add(msg.split());
+						msg.close();
```

Comment:
What do you think of overriding `encode()` method, thus `encodeAndClose()` will close the message for us?

Reply:
yes ! much better; I just committed the proposed change and I verified the memory leak is avoided on my local branch, where I have rebased this PR on top of the other pending #2484